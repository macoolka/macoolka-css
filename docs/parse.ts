
import { flatten } from 'mocoolka-fp/lib/Array';
import { readFiles } from './fs';
import { parseTs } from 'mocoolka-ast/lib/ts/interface'

export const parseProperties = (root: string) => (inputPath: string, outPath: string) => {
   // const files: DInterface[] = flatten(readFiles('**/*.ts')(root, inputPath, outPath).map(a => a.map(b => parseTs([])(b.realpath))).run());
    return flatten(readFiles('**/*.ts')(root, inputPath, outPath).map(a => a.map(b => parseTs(['CommonProperties'])(b.realpath))).run());
};