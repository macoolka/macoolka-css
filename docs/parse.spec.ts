import {parseProperties} from './parse';
import {resolve} from 'path';
describe('fs', () => {
    it('parse to object', () => {
        expect(parseProperties(resolve(__dirname,'tests'))('modules','Output')).toMatchSnapshot()

    })
})