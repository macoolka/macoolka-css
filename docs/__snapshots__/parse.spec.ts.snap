// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fs parse to object 1`] = `
Array [
  Object {
    "description": Some {
      "_tag": "Some",
      "value": "Flex properties",
    },
    "extendNames": Array [
      "SProps",
      "EProps",
    ],
    "items": Array [
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "Items should wrap or not.",
        },
        "kind": "Property",
        "name": "mkFlexWrap",
        "type": Object {
          "items": Array [
            "nowrap",
            "wrap",
            "wrap-reverse",
            "inherit",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "The direction to stack the flex items.",
        },
        "kind": "Property",
        "name": "mkFlexDirection",
        "type": Object {
          "items": Array [
            "inherit",
            "row",
            "row-reverse",
            "column",
            "column-reverse",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "Align the flex items",
        },
        "kind": "Property",
        "name": "mkFlexAlign",
        "type": Object {
          "items": Array [
            "inherit",
            "flex-start",
            "flex-end",
            "center",
            "space-between",
            "space-around",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "Align the flex items vertically.",
        },
        "kind": "Property",
        "name": "mkFlexAlignV",
        "type": Object {
          "items": Array [
            "inherit",
            "flex-start",
            "flex-end",
            "center",
            "stretch",
            "baseline",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "Align the flex lines.",
        },
        "kind": "Property",
        "name": "mkFlexAlignLines",
        "type": Object {
          "items": Array [
            "inherit",
            "flex-start",
            "flex-end",
            "center",
            "space-between",
            "space-around",
            "stretch",
            "space-evenly",
            "end",
            "start",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "Align the flex item",
        },
        "kind": "Property",
        "name": "mkFlexItemAlign",
        "type": Object {
          "items": Array [
            "inherit",
            "flex-start",
            "flex-end",
            "center",
            "stretch",
            "baseline",
            "end",
            "start",
            "self-end",
            "self-start",
            "auto",
            "left",
            "normal",
            "right",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "Align the flex item vertically.",
        },
        "kind": "Property",
        "name": "mkFlexItemAlignV",
        "type": Object {
          "items": Array [
            "inherit",
            "flex-start",
            "flex-end",
            "center",
            "stretch",
            "baseline",
            "end",
            "start",
            "self-end",
            "self-start",
            "auto",
            "normal",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "The initial length of a item.",
        },
        "kind": "Property",
        "name": "mkFlexItemWidth",
        "type": Object {
          "items": Array [
            "string",
            "number",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "The item will shrink relative to the flex items.",
        },
        "kind": "Property",
        "name": "mkFlexItemShrink",
        "type": Object {
          "items": Array [],
          "kind": "Type",
          "name": "other",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "The item will grow relative to the flex items.",
        },
        "kind": "Property",
        "name": "mkFlexItemGrow",
        "type": Object {
          "items": Array [],
          "kind": "Type",
          "name": "other",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "The order of the flex item.",
        },
        "kind": "Property",
        "name": "mkFlexItemOrder",
        "type": Object {
          "items": Array [],
          "kind": "Type",
          "name": "other",
        },
      },
    ],
    "kind": "Interface",
    "name": "Props",
    "tag": "",
  },
  Object {
    "description": Some {
      "_tag": "Some",
      "value": "Text properties",
    },
    "extendNames": Array [
      "EProps",
      "SProps",
    ],
    "items": Array [
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "font weight",
        },
        "kind": "Property",
        "name": "mkFontWeight",
        "type": Object {
          "items": Array [
            "thin",
            "light",
            "regular",
            "medium",
            "bold",
            "black",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "font family",
        },
        "kind": "Property",
        "name": "mkFontFamily",
        "type": Object {
          "items": Array [
            "sansSerif",
            "serif",
            "monospace",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "font size",
        },
        "kind": "Property",
        "name": "mkFontSize",
        "type": Object {
          "items": Array [
            "h1",
            "h2",
            "h3",
            "h4",
            "h5",
            "h6",
            "subtitle",
            "p",
            "caption",
            "overline",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "text wrap",
        },
        "kind": "Property",
        "name": "mkTextNoWrap",
        "type": Object {
          "items": Array [
            "false",
            "true",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "text transform",
        },
        "kind": "Property",
        "name": "mkTextTransform",
        "type": Object {
          "items": Array [
            "none",
            "capitalize",
            "uppercase",
            "lowercase",
            "inherit",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "text align",
        },
        "kind": "Property",
        "name": "mkTextAlign",
        "type": Object {
          "items": Array [
            "inherit",
            "left",
            "right",
            "center",
            "justify",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "text direction",
        },
        "kind": "Property",
        "name": "mkTextDirection",
        "type": Object {
          "items": Array [
            "inherit",
            "ltr",
            "rtl",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "text decoration",
        },
        "kind": "Property",
        "name": "mkTextDecoration",
        "type": Object {
          "items": Array [
            "overline",
            "none",
            "inherit",
            "underline",
            "line-through",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "text underline",
        },
        "kind": "Property",
        "name": "mkTextUnderlined",
        "type": Object {
          "items": Array [
            "false",
            "true",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "text with strong style",
        },
        "kind": "Property",
        "name": "mkTextStrong",
        "type": Object {
          "items": Array [
            "false",
            "true",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
      Object {
        "description": Some {
          "_tag": "Some",
          "value": "text with italic style",
        },
        "kind": "Property",
        "name": "mkTextItalic",
        "type": Object {
          "items": Array [
            "false",
            "true",
          ],
          "kind": "Type",
          "name": "union",
        },
      },
    ],
    "kind": "Interface",
    "name": "Props",
    "tag": "",
  },
]
`;
